packages/ra-ui-materialui/src/input/SelectArrayInput.tsx

```diff
@@ -209,7 +220,8 @@ export const SelectArrayInput = (props: SelectArrayInputProps) => {
                     disabled={getDisableValue(choice)}
                 >
                     {renderMenuItemOption(
-                        !!createItem && choice?.id === createItem.id
+                        // eslint-disable-next-line eqeqeq
+                        !!createItem && choice?.id == createItem.id
```

Comment:
same,

Besides, some of the code you changed is related to the ability to create a new item. I believe this should be covered by tests too then.

Reply:
Left as was

Reply:
For the record, I advise **against** these changes if they are not properly tested.
Ping @fzaninotto